astropy/io/fits/connect.py

```diff
@@ -302,3 +306,202 @@ def write_table_fits(input, output, overwrite=False):
 io_registry.register_reader('fits', Table, read_table_fits)
 io_registry.register_writer('fits', Table, write_table_fits)
 io_registry.register_identifier('fits', Table, is_fits)
+
+
+def read_data_fits(filename, ext_data=0, ext_meta=0, ext_mask='mask',
```

Comment:
I wonder if this name should be more specific, something like `read_nddata_fits` or even `read_image_fits` since this is really specific to 2D image data.